src/main/java/mServer/crawler/sender/arte/ArteVideoDetailsDeserializer.java

```diff
@@ -0,0 +1,194 @@
+                                    break;
+                                case BROADCASTTTYPE_MINOR_RE:
+                                    broadcastBeginMinor = value;
+                                    break;
+                                default:
+                                    LOG.debug("New broadcasttype: " + type);
+                            }
+                        }
+                    }
+                }
+
+                if(!broadcastBeginFirst.isEmpty()) {
+                    detailsDTO.setBroadcastBegin(broadcastBeginFirst);
+                } else if(!broadcastBeginMajor.isEmpty()) {
+                    detailsDTO.setBroadcastBegin(broadcastBeginMajor);
+                } else if(!broadcastBeginMinor.isEmpty()) {
+                    detailsDTO.setBroadcastBegin(broadcastBeginMinor);
+                }
+                
+                // wenn kein Ausstrahlungsdatum vorhanden, dann die erste Ausstrahlung nehmen
+                // egal, wann die CatchupRights liegen, damit ein "sinnvolles" Datum vorhanden ist
+                if(detailsDTO.getBroadcastBegin().isEmpty()) {
+                    detailsDTO.setBroadcastBegin(getFirstBroadcastDateIgnoringCatchupRights(broadcastArray));
+                }
+            } else {
+                // keine Ausstrahlungen verfügbar => catchupRightsBegin verwenden
+                JsonElement elementBegin = programElement.get(JSON_ELEMENT_BROADCAST_CATCHUPRIGHTS_BEGIN);
+                
+                if(!elementBegin.isJsonNull()) {
+                    detailsDTO.setBroadcastBegin(elementBegin.getAsString());
+                }
+            }
+        }
+            
+        return detailsDTO;
+    }
+
+    /**
+     * Liefert den Beginn der Ausstrahlung, 
+     * wenn 
+     *  - heute im Zeitraum von CatchUpRights liegt 
+     *  - oder heute vor dem Zeitraum liegt
+     *  - oder CatchUpRights nicht gesetzt ist und die Ausstrahlung in der Vergangenheit liegt
+     * @param broadcastObject 
+     * @return der Beginn der Ausstrahlung oder ""
+     */
+    private String getBroadcastDate(JsonObject broadcastObject) {
+        String broadcastDate = "";
+        
+        JsonElement elementBegin = broadcastObject.get(JSON_ELEMENT_BROADCAST_CATCHUPRIGHTS_BEGIN);
+        JsonElement elementEnd = broadcastObject.get(JSON_ELEMENT_BROADCAST_CATCHUPRIGHTS_END);
+        
+        if (!elementBegin.isJsonNull() && !elementEnd.isJsonNull()) {
+            String begin = elementBegin.getAsString();
+            String end = elementEnd.getAsString();
+
+            try {
+                Calendar beginDate = Calendar.getInstance();
+                beginDate.setTime(broadcastDateFormat.parse(begin));
+                Calendar endDate = Calendar.getInstance();
+                endDate.setTime(broadcastDateFormat.parse(end));
+
+                if(DateWithoutTimeComparer.compare(today, beginDate) >= 0 && DateWithoutTimeComparer.compare(today, endDate) <= 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this conditional structure or edit its code blocks so that they're not all the same. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3923)